pkg/virtual/initializingworkspaces/builder/build.go

```diff
@@ -115,7 +113,7 @@ func BuildVirtualWorkspace(
 			completedContext = dynamiccontext.WithAPIDomainKey(completedContext, apiDomain)
 			return true, prefixToStrip, completedContext
 		}),
-		Authorizer: newAuthorizer(kubeClusterClient),
+		Authorizer: newCachingAuthorizer(kubeClusterClient),
```

Comment:
Can the cache be shared here across all the BuildVirtualWorkspace calls?

Reply:
In this package, I'd say yes

Reply:
Perfect, updated!